### ISSUE TITLE
Loss Tensor and Summaries KerasModel

### ISSUE THREAD

**Issue by vsomnath** [deepchem/deepchem#1632]
https://github.com/deepchem/deepchem/blob/2417df0f3b129e193e47acac1d9c06341b935f40/deepchem/models/keras_model.py#L403

Shouldn't this line be:
```
fetches = [train_op, loss_tensor, self._global_step] 
```
`loss_tensor` is changed depending on whether the user specifes a loss or not.

https://github.com/deepchem/deepchem/blob/2417df0f3b129e193e47acac1d9c06341b935f40/deepchem/models/keras_model.py#L245

This should also be changed to `loss_tensor` instead of `self._loss_tensor`. 

I am currently working on a PR that tries to unify both train and validation loss computation. These issues have been fixed there.

**Comment by peastman** [comment]
> Shouldn't this line be:

Yes, I believe you're correct.  It will need a few other changes too: as currently written, `loss_tensor` won't necessarily have been created at that point.

> This should also be changed to loss_tensor instead of self._loss_tensor.

This one should not be changed.  `_create_training_ops()` is only executed once, and It creates top level fields of the object.  The things it creates should not refer to a custom loss function from one particular call to `fit()`.

**Comment by VIGS25** [comment]
> This one should not be changed. _create_training_ops() is only executed once, and It creates top level fields of the object. The things it creates should not refer to a custom loss function from one particular call to `fit()`.

Ah, I see. I was under the impression that the user has the option to overwrite the loss function used, and hence all TensorBoard logging should be based on that.

**Comment by peastman** [comment]
Possibly we do want to change what value gets logged to Tensorboard.  But the op computed in that particular line is for the standard loss.  All of the Tensorboard support is very limited at the moment.
